cmd/symbol-check/check_symbol.go

```diff
@@ -0,0 +1,348 @@
+/*symbol-check
+
+this program checks for incompatible symbols(extra exported symbols and incompatible type definitions) that might break forward compatibility when built as a plugin.
+
+Discussion at https://github.com/gotify/server/issues/51#issuecomment-452954279
+
+Sample usage:
+$ go run github.com/gotify/plugin-api/cmd/symbol-check > export_ref_do_not_edit.json # take a snapshot of the current export in every major release
+$ go run github.com/gotify/plugin-api/cmd/symbol-check -c export_ref_do_not_edit.json # compare current version for incompatible definitions
+*/
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+const pluginPkgName = "plugin"
+
+var workDir string
+var compareTo string
+
+type SymbolList []Symbol
+
+func compareSymbolList(source, target SymbolList, cmpLabel bool) []string {
+	diffs := make([]string, 0)
+
+	agg := make(map[string]*Symbol)
+	for _, symbol := range source {
+		sym := symbol
```

Comment:
this seems not very useful (:.

Reply:
The `sym` variable here is used to make a copy the symbol variable. Otherwise the all `&symbols` will point to the value of `symbol` in the final iteration.